### DISCUSSION TITLE
[Bug]: Inconsistent data in validate method of register and handleSubmit

### DISCUSSION THREAD

**Discussion by MissedWarrior** [react-hook-form/react-hook-form#6253] [Q&A] [answered]
### Version Number

6.15.8

### Codesandbox/Expo snack

https://stackblitz.com/edit/react-ts-smmudv?file=index.tsx

### Steps to reproduce

1. Open sandbox
2. Click on submit button
3. Look at console: you will see two outputs: from validation and from submit
4. Data in validation method is not equal to data in handleSubmit

### Expected behaviour

Data in validation method should be same as in handleSubmit

Tbh, it is not really a problem that data is inconsistent but there is a question then: how to validate something when the output is object? I placed validation method on input because of similarity of the situation with radio but this solution didn't solve my problem :(

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
<img width="1120" alt="Screen Shot 2021-08-17 at 8 33 11 pm" src="https://user-images.githubusercontent.com/10513364/129711115-3a4b99a5-ac3b-4039-839e-f65720f12e24.png">

couldn't see anything. I would suggest you build a controlled component and host your own state inside, that way the data would be more consistent.

**Comment by MissedWarrior** [comment]
Recreated in codesandbox. I hope this time it will work https://codesandbox.io/s/quizzical-gagarin-1yof7?file=/src/App.tsx

**Reply by bluebill1049** [reply]
consider to use `getValues()` inside the `validate` function

**Reply by MissedWarrior** [reply]
Already tried this but got the same output as in validate function argument. Only watch gives actual output

**Reply by bluebill1049** [reply]
<input name={`${name}[${key}]`} ref={register} type="checkbox" />

try without the same name, and otherwise, build a controlled component and host your state for the input like I suggested earlier, or consider to move to v7.

**Reply by MissedWarrior** [reply]
Couldn't move to v7 because of IE11. So the Controller is my solution. Anyway thank you very much!

**Reply by bluebill1049** [reply]
thanks for understanding.
